views/pages/AccountPage.jsx

```diff
@@ -1,130 +1,70 @@
-import React, { Component } from 'react'
-
+import React, { useState, useRef } from 'react'
+import classnames from 'classnames'
+import { useOutletContext } from 'react-router-dom'
 import TextField from '@material-ui/core/TextField'
 import Button from '@material-ui/core/Button'
 import Tooltip from '@material-ui/core/Tooltip'
 import ButtonBase from '@material-ui/core/ButtonBase'
 import Snackbar from '@material-ui/core/Snackbar'
 
+import api from '../api'
+import { EMAIL_REGEX } from '../../constants'
+import Form from '../forms/Form'
 import getAvatar from '../fe-utils/avatarUtil'
-import getCounts from '../fe-utils/countUtil'
-import { fetchSubjects } from '../fe-utils/fetchUtil'
 
-import { apiRoutes } from '../routes/routes'
+const AccountPage = () => {
+  const icon = 'icon'
+  const canvasRef = useRef(null)
+  const profileImageRef = useRef(null)
+  const [snackBar, setSnackBar] = useState({ open: false, message: '' })
+  const { user, classes, setUser } = useOutletContext()
 
-class AccountPage extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      user: {},
-      _id: '',
-      uid: '',
-      display_name: '',
-      title: '',
-      department: '',
-      company: '',
-      mail: '',
-      ldap: '',
-      icon: '',
-      totalStudies: 0,
-      totalSubjects: 0,
-      totalDays: 0,
-      snackTime: false,
-      mobileOpen: false,
-    }
-  }
-  async componentDidMount() {
-    try {
-      const acl = await fetchSubjects()
-      this.setState(getCounts({ acl }))
-      this.fetchUserInfo(this.props.user.uid)
-      this.setState({
-        user: this.props.user,
-      })
-    } catch (err) {
-      console.error(err.message)
-    }
-  }
-  componentWillMount() {
-    this.fetchUserInfo(this.props.user.uid)
-  }
-  handleDrawerToggle = () => {
-    this.setState((state) => ({ mobileOpen: !state.mobileOpen }))
-  }
-  fetchUserInfo = (uid) => {
-    return window
-      .fetch(apiRoutes.users.user(uid), {
-        method: 'GET',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        credentials: 'same-origin',
-      })
-      .then((response) => {
-        if (response.status !== 200) {
-          return
+  const handleChange = (e) => {
```

Comment:
It might be clearer to have one function for updating files and another for making other form data updates. Easier from a testing perspective at least.

Reply:
[0ca9ade](https://github.com/AMP-SCZ/dpdash/pull/320/commits/0ca9ade589cf737b00589944549c08cd5fa728bc)